numpy/f2py/tests/util.py

```diff
@@ -345,12 +365,41 @@ def setup(self):
 
         # Build the module
         if self.code is not None:
-            self.module = build_code(self.code, options=self.options,
-                                     skip=self.skip, only=self.only,
-                                     suffix=self.suffix,
-                                     module_name=self.module_name)
+            self.module = build_code(
+                self.code,
+                options=self.options,
+                skip=self.skip,
+                only=self.only,
+                suffix=self.suffix,
+                module_name=self.module_name,
+            )
 
         if self.sources is not None:
-            self.module = build_module(self.sources, options=self.options,
-                                       skip=self.skip, only=self.only,
-                                       module_name=self.module_name)
+            self.module = build_module(
+                self.sources,
+                options=self.options,
+                skip=self.skip,
+                only=self.only,
+                module_name=self.module_name,
+            )
+
+
+#
+# Helper functions
+#
+
+
+def getpath(*a):
+    # Package root
+    d = Path(numpy.f2py.__file__).parent.resolve()
```

Comment:
I confess that I cannot see why this works, but it seems to do.

Reply:
@charris could you be more specific with this concern? I am curious if I might miss something here.

Reply:
@pearu I couldn't get it to work for me without first importing `numpy.f2py`, importing `numpy` alone led to an `AttributeError` (IIRC).  I'm guessing the import happens somewhere along the line.

Reply:
Ah, ok then. Indeed, for this to work, one must have `import numpy`.
Also, `import numpy` does not trigger `import numpy.f2py` and an AttributeError is expected.
However, it works here because we are inside `numpy.f2py.tests.util` module that implies that `numpy.f2py` has been important while getting there. I hope my explanation makes sense.

Reply:
I think it comes in when `F2PyTest.setup` is run by pytest. `F2PyTest.setup` then  runs `build_module` which does the import. I think it might be safer to do the import directly in `getpath`. Note that `util.getpath` in used in classes based on `F2PyTest`.

Reply:
But it might also be the case that pytest instantiates the test classes in an environment containing `numpy.f2py`.